docs/specification/objects.md

```diff
@@ -29,6 +29,30 @@ The Properties, Parameters and Conditions of Object Templates and of Objects are
 An [Object](../objects) can have [Attributes](../attributes) that are not defined in its [Object Template](../object-templates), and an Object Template can have Attributes that are not defined in associated Objects.
 However, an Object cannot have two attributes of the same type (Property, Parameter, or Condition) with the same name or the same attribute template.
 
+#### Material History
+
+A material history is the complete set of objects that precede a [Material Run](#material-run) in time, as well as the associated Specs and Templates.
+That final or ultimate material is called the `Terminal Material`.
+Given a dataset of relevant objects, a Material History would be collected by choosing a Terminal Material and then recursively:
+* extracting the process that generated it
+* extracting the ingredients that went into that process
+* extracting the materials that constituted those ingredients
+until one reaches a point that a process has no recorded ingredients (e.g., I purchased a bag of flour).
+The dataset itself may have a complicated web of relationships (e.g., I made 12 different loaves of bread from one bag of flour) and the material history is just one example of a way to consider that dataset.
+
+Which materials would be terminal depend on context.
+A steelmaker might consider a steel plate to be terminal in their process.
+A machinist might take that plate as input for the dog bone they manufacture as terminal in their process.
+A technician might take that dog bone as input to destructive testing, and then a scrapper would feed the broken dog bone back as feedstock for another steelmaker.
+Every history has one terminal material, and a given dataset will likely have several terminal materials in that context.
```

Comment:
What is `that context`?

Reply:
`that context` == the dataset.  In a set of objects, a subset will be Material Runs that are not linked by Ingredient Runs.  They may be referenced in other datasets, and so are only terminal in the context of the first dataset.

Reply:
But datasets aren't part of GEMD.  Its not clear to me if this is saying that "the run at the end of the material history is called the terminal material" (as in "That final or ultimate material is called the `Terminal Material`.") or "the material runs that are not referenced within a dataset are terminal materials in the scope of that dataset" (which is a definition that I've heard thrown around internally but doesn't make sense to me here as part of the GEMD specification).

Reply:
The access control concept `Dataset` is not part of GEMD.  However, the point about the Steelmaker-Machinist-Technician-Scrapper bit is that each might narrowly consider their own context, or consider a broader context, and the identity of the last material in the chain follows from that perspective.

If you'd prefer a different term for clarity or an explicit definition, that's fine.

Reply:
I'm fine with the term.  How about just getting rid of `, and a given dataset will likely have several terminal materials in that context` ?  I think your Steelmaker-Machinist-Technician-Scrapper is enough as is and avoids any confusion with Citrine's "Dataset" term.